internal/common/rpc.go

```diff
@@ -95,17 +95,17 @@ func (rpc *RPC) checkSupportedMethods() error {
 	var getLogsResult interface{}
 	logsErr := rpc.RPCClient.Call(&getLogsResult, "eth_getLogs", map[string]string{"fromBlock": "0x0", "toBlock": "0x0"})
 	if logsErr != nil {
-		return fmt.Errorf("eth_getBlockByNumber method not supported: %v", logsErr)
+		return fmt.Errorf("eth_getLogs method not supported: %v", logsErr)
 	}
 	log.Debug().Msg("eth_getLogs method supported")
 
+	var traceBlockResult interface{}
```

Comment:
curious why you check for this even when tracing is disabled

Reply:
I am not. It is making a trace request only if configuration allows it. 
But if it's allowed we still need to check if the rpc allows it so we make a request.
That's why we can't solely rely on the config